DCM-22: Support Financial July period type in automation

Piumal1999 opened the pull request:
## The issue I worked on
See [https://issues.openmrs.org/browse/DCM-22](https://issues.openmrs.org/browse/DCM-22) 

## Description of what I changed:
Add Financial July period type support for automation by updating the following method:
```src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java:transformToDHISPeriod```

jayasanka-sack reviewed (changes requested):
This will run the report for an incomplete year if today's date is between Jan and June.

Ex: If today is the 1st of March 2020, then the start date will be 1st July 2019 and the end date will be 30th June 2020 (which is a future date)

Piumal1999 commented:
> This will run the report for an incomplete year if today's date is between Jan and June.
> 
> Ex: If today is the 1st of March 2020, then the start date will be 1st July 2019 and the end date will be 30th June 2020 (which is a future date)

Oh sorry i didn't notice it previously. Thank you.

Added an if statement to set the year. And fixed the issue.

jayasanka-sack reviewed (approved):
Thanks @Piumal1999 !
